### ISSUE TITLE
Some tests use fixtures whith scopes that are at odds with hypothesis

### ISSUE THREAD

**Issue by jbarnoud** [MDAnalysis/mdanalysis#2985]
## Expected behavior ##

Running the tests does not throw warnings.

## Actual behavior ##

Hypothesis warns about the scope of some fixture potentially causing issues with generated test cases.

```
MDAnalysisTests/formats/test_libdcd.py::test_written_remarks_property
  /home/jon/miniconda3/envs/py38/lib/python3.8/site-packages/hypothesis/extra/pytestplugin.py:172: HypothesisDeprecationWarning: MDAnalysisTests/formats/test_libdcd.py::test_written_remarks_property uses the 'tmpdir' fixture, which is reset between function calls but not between test cases generated by `@given(...)`.  You can change it to a module- or session-scoped fixture if it is safe to reuse; if not we recommend using a context manager inside your test function.  See https://docs.pytest.org/en/latest/fixture.html#sharing-test-data for details on fixture scope.
    note_deprecation(

MDAnalysisTests/formats/test_libdcd.py::test_written_remarks_property
  /home/jon/miniconda3/envs/py38/lib/python3.8/site-packages/hypothesis/extra/pytestplugin.py:172: HypothesisDeprecationWarning: MDAnalysisTests/formats/test_libdcd.py::test_written_remarks_property uses the 'dcd' fixture, which is reset between function calls but not between test cases generated by `@given(...)`.  You can change it to a module- or session-scoped fixture if it is safe to reuse; if not we recommend using a context manager inside your test function.  See https://docs.pytest.org/en/latest/fixture.html#sharing-test-data for details on fixture scope.
    note_deprecation(
```

## Code to reproduce the behavior ##

`pytest formats/test_libdcd.py`

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0-dev0 64a7c051f1a51ff2a89b9da9257a21cea0ef00fa
- Which version of Python (`python -V`)? Python 3.8.5
- Which operating system? Ubuntu 20.04 on WSL

**Comment by lilyminium** [comment]
This now reliably fails.

See: https://github.com/MDAnalysis/mdanalysis/pull/3091/checks?check_run_id=1674311671#step:10:596

and https://github.com/MDAnalysis/mdanalysis/pull/2664/checks?check_run_id=1715357196#step:10:296

```
/test_libdcd.py::test_written_remarks_property uses the 'tmpdir' fixture, which is reset between function calls but not between test cases generated by `@given(...)`.  You can change it to a module- or session-scoped fixture if it is safe to reuse; if not we recommend using a context manager inside your test function.  See docs.pytest.org/en/latest/fixture.html#sharing-test-data for details on fixture scope.
E                               See hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.function_scoped_fixture to the suppress_health_check settings for this test.
```

**Comment by IAlibay** [comment]
If I'm understanding the error, all we have to do is use `tmpdir.as_cwd`?
